### PR TITLE
Add support for wolfSSL_CTX_load_system_CA_certs on Windows and Mac.

### PR DESCRIPTION
**Pull request by haydenroche5**
I ran the example client in Visual Studio with `-h google.com -p 443` and `-h google.com -p 443 --sys-ca-certs`. The former fails, as expected, and the latter succeeds, indicating system CA certs were loaded in order to validate google.com's cert chain. Ran the same test on macOS.

Adds `WOLFSSL_SYS_CA_CERTS` option for feature.

### CODE REVIEW

File: tests/api.c [hunk 1]
```diff
@@ -1352,31 +1352,24 @@ static int test_wolfSSL_CTX_load_system_CA_certs(void)
...
-#else
+    #ifdef _WIN32
+        dirValid = 1;
+    #else
         word32 numDirs;
         const char** caDirs = wolfSSL_get_system_CA_dirs(&numDirs);
 
         if (caDirs == NULL || numDirs == 0) {
-            ret = -1;
+            ret = -2;
```
Comment (dgarske) [inline tests/api.c:1369]: Should we come up with better return codes than `-1` through `-4`?

File: tests/api.c [hunk 2]
```diff
@@ -1352,31 +1352,24 @@ static int test_wolfSSL_CTX_load_system_CA_certs(void)
...
-        if (wolfSSL_CTX_load_system_CA_certs(ctx) != WOLFSSL_NOT_IMPLEMENTED) {
-            ret = -1;
-        }
-#ifdef OPENSSL_EXTRA
-        if (wolfSSL_CTX_set_default_verify_paths(ctx) != WOLFSSL_FAILURE) {
-            ret = -1;
-        }
-#endif /* OPENSSL_EXTRA */
-#else
+    #ifdef _WIN32
```
Comment (dgarske) [inline tests/api.c:1362]: You might consider using `USE_WINDOWS_API`. Some targets report `_WIN32`, but don't really support "windows".

**Review by dgarske — CHANGES_REQUESTED** [review]

File: tests/api.c (hunk 2 above)
Comment (haydenroche5) [reply tests/api.c:1362]: Sure. I also realized we don't actually build api.c for Windows, as far as I can tell from looking at our VS solutions.

File: tests/api.c (hunk 1 above)
Comment (haydenroche5) [reply tests/api.c:1369]: The idea is to correlate the failure with a specific line for debug. It seems like overkill, IMO, to give them named constants for such a small test. But open to suggestions.

Comment (haydenroche5) [reply tests/api.c:1369]: Added `fprintf` statements to tell the user which function failed.

**Comment by haydenroche5** [comment]
Added Mac support to this PR.

File: configure.ac [hunk 3]
```diff
@@ -8001,6 +7999,10 @@ case $host_os in
                 MINGW_LIB_WARNING="yes"
             fi
         fi ;;
+    *darwin*)
+        # For Mac we need these frameworks to load system CA certs
+        LDFLAGS="$LDFLAGS -framework CoreFoundation -framework Security"
```
Comment (dgarske) [inline configure.ac:8004]: Does this also need added to CMakeList.txt?

```
[ 61%] Linking C shared library libwolfssl.dylib
Undefined symbols for architecture x86_64:
  "_CFArrayGetCount", referenced from:
      _LoadSystemCaCertsMac in ssl.c.o
  "_CFArrayGetValueAtIndex", referenced from:
      _LoadSystemCaCertsMac in ssl.c.o
  "_CFDataGetBytePtr", referenced from:
      _LoadSystemCaCertsMac in ssl.c.o
  "_CFDataGetLength", referenced from:
      _LoadSystemCaCertsMac in ssl.c.o
  "_CFRelease", referenced from:
      _LoadSystemCaCertsMac in ssl.c.o
  "_SecCertificateCopyData", referenced from:
      _LoadSystemCaCertsMac in ssl.c.o
  "_SecTrustSettingsCopyCertificates", referenced from:
      _LoadSystemCaCertsMac in ssl.c.o
ld: symbol(s) not found for architecture x86_64
```

**Review by dgarske — CHANGES_REQUESTED** [review]
Otherwise this works great and looks great! Thank you! This is an awesome feature!

File: CMakeLists.txt [hunk 4]
```diff
@@ -1526,6 +1538,10 @@ endif()
 
 # TODO: - Fast huge math
 
+if("${CMAKE_SYSTEM_PROCESSOR}" STREQUAL "x86_64")
```
Comment (haydenroche5) [inline CMakeLists.txt:1541]: This was missing and causing us to fail to verify the signature on one of the certs in google.com's chain.

**Review by dgarske — CHANGES_REQUESTED** [review]
This is all awesome and looks great! Can you please add a build option to allow compiling out this feature for customers who have code size or compatibility issues? Something like ‘#ifndef WOLFSSL_NO_SYSTEM_CERTS’?

**Review by dgarske — APPROVED** [review]

**Pull request merged**